src/Workspaces/CSharp/Portable/CodeGeneration/EventGenerator.cs

```diff
@@ -199,6 +200,11 @@ private static BlockSyntax GenerateBlock(IMethodSymbol accessor)
                         tokens.Add(SyntaxFactory.Token(SyntaxKind.StaticKeyword));
                     }
 
+                    if (@event.AddMethod?.IsReadOnly == true)
```

Comment:
Does the RemoveMethod need to be checked? I can guess your answer, but it might be worth writing it in a source comment here. Otherwise this looks like an odd ommission. Should IEventSymbols just have IsReadOnly that implements this internally so we're also not embedding metadata rules?

Reply:
Yeah, an event from metadata is readonly iff its accessors are readonly. Will add a comment. Tracking the public API discussion in #34213